include/capmap-mappings.h

```diff
@@ -93,7 +93,65 @@ class LoadMap : public Map {
   SparseRange ranges_;
 };
 
-// TODO: Provide other `Map` derivatives.
+// Ranges with given permissions
+class BasicMap : public Map {
+ public:
+  BasicMap(const char *name, const char *addrsp, size_t perms)
+      : name_(name), addrsp_(addrsp), perms_(perms) {}
+  virtual char const *name() const override { return name_; }
+  virtual char const *address_space() const override { return addrsp_; }
+  virtual std::set<Range> const &ranges() const override { return ranges_.parts(); }
+  virtual bool try_combine(void *__capability cap) override;
+  virtual ~BasicMap() {}
+
+  SparseRange const &sparse_range() const { return ranges_; }
+
+ private:
+  SparseRange ranges_;
+  const char *const name_;
+  const char *const addrsp_;
+  const size_t perms_;
+};
+
+// Memory ranges which can be branched to
+class BranchMap : public Map {
+ public:
+  virtual char const *name() const override { return "branch"; }
+  virtual char const *address_space() const override { return "virtual memory"; }
+  virtual std::set<Range> const &ranges() const override { return ranges_.parts(); }
+  virtual bool try_combine(void *__capability cap) override;
+  virtual ~BranchMap() {}
+
+  SparseRange const &sparse_range() const { return ranges_; }
+
+ private:
+  SparseRange ranges_;
+};
+
+typedef bool (*poison_callback_t)(void *__capability cap);
```

Comment:
Just a suggestion. 
I would add `void * user_data` to the callback, in case some context needs to be passed in.